pkg/handlers/v1/eni.go

```diff
@@ -0,0 +1,115 @@
+package v1
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+const elbManaged = "amazon-elb"
+
+type eniConfiguration struct {
+	Description        string      `json:"description"`
+	PrivateIPAddresses []privateIP `json:"privateIpAddresses"`
+	RequesterID        string      `json:"requesterId"`
+	RequesterManaged   bool        `json:"requesterManaged"`
+}
+
+type eniConfigurationDiff struct {
+	PreviousValue *eniConfiguration `json:"previousValue"`
+	UpdatedValue  *eniConfiguration `json:"updatedValue"`
+	ChangeType    string            `json:"changeType"`
+}
+
+type privateIP struct {
+	PrivateIPAddress string `json:"PrivateIpAddress"`
+}
+
+type eniTransformer struct{}
+
+func (t eniTransformer) Create(event awsConfigEvent) (Output, error) {
+	output, err := getBaseOutput(event.ConfigurationItem)
+	if err != nil {
+		return Output{}, err
+	}
+
+	var config eniConfiguration
+	if err := json.Unmarshal(event.ConfigurationItem.Configuration, &config); err != nil {
+		return Output{}, err
+	}
+
+	if filter(config) {
+		return output, nil
+	}
+
+	change := extractEniInfo(&config)
+	change.ChangeType = added
+	output.Changes = append(output.Changes, change)
+
+	return output, nil
+}
+
+// Returns true if we should filter this event due to not being requester managed
```

Comment:
question: what does requester managed mean?

Reply:
The requester in this case is an AWS resource. I've been calling it that based on https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/requester-managed-eni.html

Our specific use case is around when Load Balancers request an ENI for themselves that they then manage, but other AWS resources can do this as well. This is also why we have checks for the requesterId, to make sure the ENI that we're saving is attached to a load balancer.